src/main/java/duke/filehandler/Storage.java

```diff
@@ -0,0 +1,122 @@
+package duke.filehandler;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+import java.util.Scanner;
+public class Storage {
+
+    private static String FILE_PATH;
+    public Storage (String file_path){
+        FILE_PATH = file_path;
+    }
+
+    /***
+     * converts string from saved file to Date object
+     * @param dateString
+     * @return Date
+     * @throws DukeException
+     */
+    public static Date fileDateParser(String dateString) throws DukeException {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd MMM yyyy h a");
+
+        try {
+            Date date = sdf.parse(dateString);
+            return date;
+        } catch (Exception e) {
+            System.out.println(e);
+           throw new DukeException("Cannot read time from file");
```

Comment:
There appears to be an indentation error here:
```suggestion
            throw new DukeException("Cannot read time from file");
```